src/main/java/nl/knaw/dans/filemigration/api/ActualFile.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (C) 2021 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.filemigration.api;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.IdClass;
+import javax.persistence.Table;
+import java.util.Objects;
+
+@Entity
+@IdClass(ActualFileKey.class)
+@Table(name = "actual")
+public class ActualFile {
+  // https://docs.jboss.org/hibernate/orm/5.6/userguide/html_single/Hibernate_User_Guide.html#schema-generation
+
+  public ActualFile() {}
+
+  public ActualFile(String doi, String actual_path, int version_nr, String sha1_checksum, String storage_id) {
+    this.doi = doi;
+    this.actual_path = actual_path;
+    this.version_nr = version_nr;
+    this.sha1_checksum = sha1_checksum;
+    this.storage_id = storage_id;
+  }
+
+  // most lengths from easy-dtap/provisioning/roles/easy-fs-rdb/templates/create-easy-db-tables.sql
+  // doi length as in dd-dtap/shared-code/dataverse/scripts/database/create/create_v*.sql
+
+  @Id
+  @Column(length = 255)
+  private String doi;
+
+  @Id
+  @Column(length = 1024) // TODO basic_file_meta has only 1000
+  private String actual_path;
+
+  @Id
+  @Column()
+  private int version_nr;
+
+  @Column(length = 40)
+  private String sha1_checksum = "";
+
+  @Column(length = 60)
+  private String storage_id = "";
+
+  public int getVersion_nr() {
+    return version_nr;
+  }
+
+  public void setVersion_nr(int version_nr) {
+    this.version_nr = version_nr;
+  }
+
+  public void incRemoved_duplicate_file_count() {
+    this.version_nr += 1;
```

Comment:
what's this?

Reply:
used on retries for duplicates https://github.com/DANS-KNAW/dd-verify-file-migration/blob/e3e5b9d23a2d51a54ef724eae661000dae966fbd/src/main/java/nl/knaw/dans/filemigration/core/EasyFileLoader.java#L76

Reply:
i still don't understand. isn't this the version as retrieved from Dataverse? why incrementing it? I understand this for the Expected file (there you don't increment the version, right?). But for the actual, a duplicate should produce an error. how can there be duplicate files in Dataverse? let's not fake a version that is not actually in Dataverse

Reply:
copy-paste devil from expected